airflow/models/dagrun.py

```diff
@@ -863,7 +859,8 @@ def verify_integrity(self, session: Session = NEW_SESSION):
                 else:
                     self.log.info("Restoring mapped task '%s'", ti)
                     Stats.incr(f"task_restored_to_dag.{dag.dag_id}", 1, 1)
-                    ti.state = State.NONE
+                    existing_indexes[task].append(ti.map_index)
+                    expected_indexes[task] = list(range(num_mapped_tis))
```

Comment:
The second line seems to overwrite the first?

Reply:
I don’t think so. One is `expected_indexes` while the other is `existing_indexes`. Maybe I’m missing something?

Reply:
Got it. My thinking was that the length won’t change for task instances of a mapped task. I’ve thought of updating but it seems no need. WDYT?